src/main.py

```diff
@@ -48,42 +57,118 @@ async def root():
 # Could look at 'get_forecast_metadata' to see when this forecast is made
 
 # get_sites: Clients get the site id that are available to them
-@app.get("/sites/site_list")
-async def get_sites(pv_sites: PV_Sites):
-    return {"pv_sites": PV_Sites}
+@app.get("/sites/site_list", response_model=PV_Sites)
+async def get_sites(site_uuid: str):
+    # simple 2. (fake just return a list of one site using 'fake_site_uuid'
+    pv_site = PV_Site_Metadata(
+        uuid=site_uuid,
+        site_name="fake site name", 
+        latitude= 50,
+        longitude= 0,
+        capacity_kw= 1 
+    )
+    pv_site_list = PV_Sites(
+        site_list= [pv_site],
+    )
+
+    return pv_site_list
 
 
 # # post_pv_actual: sends data to us, and we save to database
-@app.post("/sites/pv_actual/{site_id}")
-async def post_pv_actual(one_pv_actual: One_PV_Actual):
-    return {"one_pv_actual": One_PV_Actual}
+@app.post("/sites/pv_actual/{site_id}", response_model=One_PV_Actual)
+async def post_pv_actual(site_uuid:str, datetime_utc:str, actual_generation_kw:float):
+    # simple 4. (fake = just return what is put in)
+
+    fake_pv_input_iteration = One_PV_Actual(
+        site_uuid=site_uuid, 
+        datetime_utc=datetime_utc, 
+        actual_generation_kw=actual_generation_kw
+    )
+    
+    return fake_pv_input_iteration
 
 
 # put_site_info: client can update a site
-@app.put("/sites/pv_actual/{site_id}/info")
-async def put_site_info(site_metadata: PV_Site_Metadata):
-    return {"site_metadata": PV_Site_Metadata}
+@app.put("/sites/pv_actual/{site_id}/info", response_model=PV_Site_Metadata)
+async def put_site_info(site_uuid:str, site_name:str, latitude:int, longitude: int, capacity_kw: float  ):
```

Comment:
Same as above, we can actually pass in a `PV_Site_Metadata` model, rather than all the individual things

Reply:
See example here - https://betterprogramming.pub/why-you-should-use-a-put-request-instead-of-a-post-request-13b593b6e67c